index.d.ts

```diff
@@ -0,0 +1,36 @@
+interface Options {
```

Comment:
Don't you need to export the interface? I'm really confused about when you have to export it as I see both ways being used in other projects

Reply:
You only need to export it if you want that others can import it. Currently it's not possible to do `import {Options} from 'p-map';`. If you export the interface, then it's possible to do that.

Reply:
When would someone need to import it though? Like how do you decide when to export or not? I'm still confused about that that. For example, in Chalk, it's exported: https://github.com/chalk/chalk/blob/70f22d87ba4348637f236686bd17318922107a44/types/index.d.ts#L11

Reply:
I don't think there's a general rule for this. Why would someone import it? If they want to type it in their application.

```ts
import pMap, {Options} from 'p-map';

const options: Options = {};

if (Math.random() > 0.5) {
    options.concurrency = 5;
}

pMap(urls, mapper, options);
```

This allows you to strongly type the `options` object and don't have to go with `any`. So while typing this, it might make more sense to actually export it.

A general rule that we follow might be that, everything that's used by the outside world should be exported. In this case, because `Options` is part of the exported default function, we want to export that as well so developers can strongly type that parameter.

If we want to do this, we should come up with something better then `Options`. Because this would mean that we will have 100 `Options` interfaces for all the different libraries. Maybe go with `PMapOptions` or something?